### ISSUE TITLE
Incorrect behaviour if font shorthand contain numbers (font-family-name-quotes)

### ISSUE THREAD

**Issue by hudochenkov** [stylelint/stylelint#1680]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Incorrect behaviour if `font-weight` in `font` shorthand is a number.

> Which rule, if any, is this issue related to?

`font-family-name-quotes`

> What CSS is needed to reproduce this issue?

```css
div {
	font: italic 300 16px/30px "Merriweather", serif;
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
	"rules": {
		"font-family-name-quotes": "always-unless-keyword"
	}
}
```

> Which version of stylelint are you using?

`7.0.2`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

PostCSS plugin.

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No.

> What did you expect to happen?

No errors, because font name with quotes.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warning was flagged:

```shell
2:2 ⚠  Expected quotes around "300 Merriweather" (font-family-name-quotes) [stylelint]
```

**Comment by sendilkumarn** [comment]
I think we need to create a `set` for font-weight `100` till `900`

**Comment by sendilkumarn** [comment]
There are two ways here.

1. We should add in new set with values from 100-900 and use it.
2. We can decouple font weight and check it separately

**Comment by hudochenkov** [comment]
Thank you, @sendilkumarn!
